### PR TITLE
[EN-1480] Refactor Connector lib backfill

### PR DESCRIPTION
**Pull request by cnkarz**
Refactors Connector lib to improve backfill

### CODE REVIEW

File: chain/ethereum.go [hunk 1]
```diff
@@ -6,9 +6,8 @@ import (
 	"math/big"
 	"strings"
 
-	"github.com/nakji-network/connector/common"
-
 	"github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
```
Comment (rukolahasser) [inline chain/ethereum.go:10]: Could you remove the alias?
```suggestion
	"github.com/ethereum/go-ethereum/common"
```

File: subscription.go [hunk 2]
```diff
@@ -230,7 +227,7 @@ func (s *Subscription) subscribeLogs() {
 			s.done <- true
 			return
 
-		case err = <-subErrChan:
+		case err = <-errch:
```
Comment (rukolahasser) [inline subscription.go:230]: why does it return the line 238?

File: connector.go [hunk 3]
```diff
@@ -77,7 +77,7 @@ func NewConnector() (*Connector, error) {
 
 	//	Create kafka Produce channel and provide an outlet to connector object
 	eventSink := make(chan protoreflect.ProtoMessage)
-	go c.InitProduceChannel(eventSink)
+	go c.initProduceChannel(eventSink)
```
Comment (rukolahasser) [inline connector.go:80]: I notice that the timestamp is not set when passing to `ProduceMsg()`

File: subscription.go [hunk 4]
```diff
@@ -29,37 +28,35 @@ type ISubscription interface {
...
-	client            *ethclient.Client
-	context           context.Context
-	connector         *Connector
-	done              chan bool
-	errchan           chan error
-	fromBlock         uint64
-	headers           chan *types.Header
-	interrupt         chan os.Signal
-	isHeaderRequired  bool
+	context context.Context
```
Comment (rukolahasser) [inline subscription.go:31]: [Don't add a Context member to a struct type; instead add a ctx parameter to each method on that type that needs to pass it along. The one exception is for methods whose signature must match an interface in the standard library or in a third party library.](https://github.com/golang/go/wiki/CodeReviewComments#contexts)

**Review by rukolahasser — COMMENTED** [review]
There are a lot of recursions there, could you add some tests?

File: subscription.go (hunk 2 above)
Comment (cnkarz) [reply subscription.go:230]: to terminate this goroutine

File: connector.go (hunk 3 above)
Comment (cnkarz) [reply connector.go:80]: it is in the `msg`
```c.ProducerInterface.ProduceMsg(topic, msg, nil, time.Time{}, nil)```

Comment (rukolahasser) [reply connector.go:80]: But it's always `time.Time{}`

File: subscription.go (hunk 2 above)
Comment (rukolahasser) [reply subscription.go:227]: But this part
```
if isRetryable(err) {
    s.subscribeLogs()
}
```
blocks the `return` statement

File: connector.go (hunk 3 above)
Comment (cnkarz) [reply connector.go:80]: That tells `ProduceMsg` to always use the timestamp in the `msg`.
If some other caller of `ProduceMsg` wants to set the timestamp themselves, they can send a non-zero timestamp to override the one in the `msg`.

File: subscription.go (hunk 2 above)
Comment (cnkarz) [reply subscription.go:227]: nice catch! fixed it! thanks

File: connector.go (hunk 3 above)
Comment (rukolahasser) [reply connector.go:80]: Inside the `c.initProduceChannel()`, it always passes `time.Time{}`, how would caller set the timestamp if they wanna use `connector` lib?

Comment (cnkarz) [reply connector.go:80]: They set the timestamp inside the protobuf message. If they want more customized solution, they can use kafkautils lib and `ProduceMsg` method and then they can set timestamp to an arbitrary value.

**Review by rukolahasser — APPROVED** [review]

**Pull request merged**